lib/modules/manager/gleam/artifacts.ts

```diff
@@ -0,0 +1,89 @@
+import is from '@sindresorhus/is';
+import { TEMPORARY_ERROR } from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import type { ExecOptions } from '../../../util/exec/types';
+import {
+  deleteLocalFile,
+  readLocalFile,
+  writeLocalFile,
+} from '../../../util/fs';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+
+export async function updateArtifacts(
+  updateArtifact: UpdateArtifact,
+): Promise<UpdateArtifactsResult[] | null> {
+  const { packageFileName, updatedDeps, newPackageFileContent, config } =
+    updateArtifact;
+  logger.debug(`gleam.updateArtifacts(${packageFileName})`);
+  const isLockFileMaintenance = config.updateType === 'lockFileMaintenance';
+
+  if (is.emptyArray(updatedDeps) && !isLockFileMaintenance) {
+    logger.debug('No updated gleam deps - returning null');
+    return null;
+  }
+
+  // Find the first gleam dependency in order to handle mixed manager updates
+  const lockFileName = updatedDeps.find((dep) => dep.manager === 'gleam')
+    ?.lockFiles?.[0];
```

Comment:
isn't the lockfile always in repo root?  you should simply use the siblings fs helper to find it relative to the package file.

Reply:
sure. there is existing modeling for lockFiles in the `PackageDep` type, and I observed other modules using this. I have done what you've requested, but IMHO continuation of the `lockFiles` seems more consistent with other pkg managers? for now, new commits coming in refactoring to use getSiblingFile